monkey/common/events/abstract_event.py

```diff
@@ -21,7 +22,7 @@ class AbstractEvent(ABC):
         :param tags: The set of tags associated with the event
     """
 
-    source: UUID
-    target: Union[UUID, IPv4Address, None]
-    timestamp: float
     tags: FrozenSet[str]
+    target: Union[UUID, IPv4Address, None]
+    source: UUID = field(default_factory=getnode)
+    timestamp: float = field(default_factory=time.time)
```

Comment:
Change the parameters' order in the docstring according to this.

Reply:
Why did these get switched? Add a default emtpy `frozenset` to tags.